api/v1alpha3/condition_types.go

```diff
@@ -95,3 +95,14 @@ type Condition struct {
 type Conditions []Condition
 
 // ANCHOR_END: Conditions
+
+// CreateReplicasUpdatedCondition creates a static ReplicasUpdated Condition
+func CreateReplicasUpdatedCondition(message string) Condition {
+	return Condition{
+		Type:     ReplicasUpdated,
+		Status:   corev1.ConditionTrue,
+		Severity: ConditionSeverityInfo,
+		Reason:   ReplicasUpdatedReason,
```

Comment:
Not sure why this needs a reason since it's successful / status is True?

Reply:
why not just reuse `TrueCondition`?

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/3432/files#diff-d819bbc45f852a21035acd16bc1ff5d4R80 Specifies it cannot be empty.

I don't know what TrueCondition is.

I'm open to suggestion here if there's consensus on what these fields should be named/called, etc.  I just need the transition timestamp, I don't really care about the details.  I think it's fine as is (eg, would make some sense to the user), but I'm happy to merge something else.

Reply:
TrueCondition is https://github.com/kubernetes-sigs/cluster-api/blob/2b501499337f575540d1f6846b60f345a26f468f/util/conditions/setter.go#L80

Reply:
+1 to leave reason blank for a true condition

Reply:
This suggestion seems to be in direct contravention of the API according to the comments.  I'm not opposed to it, but if this is acceptable, we should update the comments.

Reply:
I think that just means it can't be `""`, but the field is optional. That also says `// The Severity field MUST be set only when Status=False.`

Reply:
The zero value of a string is "".  If the field is optional and we don't set it, it will evaluate to "".  I can set Severity to none or not set it, but it seems the reason should be present.

Reply:
So far, there is a nice consistency on Cluster API conditions, and this derives not only by the comments on the API fields but also by the fact that everyone uses the conditions util (that includes the TrueCondition method).

Now, trying to get to root of the problem, IMO in this PR we are tryng to stretch the idea of conditions, and this is why there are such discussion.

MachineSetReplicasUpdated does not beahve live other conditions, because it is alway status=true.
As @michaelgugino was telling above, he just need the transition timestamp, and everything else in the condition is not relevant, but neverthess, we want to make different from other conditions in order to support this different behavior.

Also, as noticed in another comments below, this PR is strectching the meaning of LastTransitionTime, that in the API is defined as the Last time the condition transitioned from one status to another. Instead, in this case the condition will be always true, but nevertheless we are triggering the change of LastTransitionTime.

This IMO will confuse also the users, because this condition is not consistent with the rest of the other conditions.

I'm starting to wander if the right approach to solve this problem is add this field to MachinesSet.Status instead of forcing this information inside a condition...

Reply:
I feel like this mostly fits the concept of a condition.  I don't feel inclined to make this condition meet some kind of purity test.  If we really want to get into the weeds, we can set a true/false condition for just 'AvailableReplicas' matching the requested amount.  This would leave a gap for scale-down operations as those will instantly match, for that we could just compare old and new counts.

I mostly envisioned this as a sort of heart beat so we know the machine-controller is doing work and prevent race conditions for other components that are relying on an up-to-date replica count.  The condition as-is allows anyone to consume any replica count (available or other) and know the information is not stale.  Maybe the message or 'reason' needs more clarity that this thing is always true.